app/models/name.rb

```diff
@@ -408,33 +408,95 @@ class Name < AbstractModel
   ### Module Name::Spelling
   scope :with_correct_spelling, -> { where(correct_spelling_id: nil) }
-
-  # For a glitch discovered in the wild:
+  scope :with_incorrect_spelling, -> { where.not(correct_spelling_id: nil) }
   scope :with_self_referential_misspelling, lambda {
     where(Name[:correct_spelling_id].eq(Name[:id]))
   }
-
+  scope :with_synonyms, -> { where.not(synonym_id: nil) }
+  scope :without_synonyms, -> { where(synonym_id: nil) }
+  ### Module Name::Create
+  # Get list of Names that are potential matches when creating a new name.
+  # Takes results of Name.parse_name.  Used by NameController#create_name.
+  # Three cases:
+  #
+  #   1. group with author       - only accept exact matches
+  #   2. nongroup with author    - match names with correct author or no author
+  #   3. any name without author - ignore authors completely when matching names
+  #
+  # If the user provides an author, but the only match has no author, then we
+  # just need to add an author to the existing Name.  If the user didn't give
+  # an author, but there are matches with an author, then it already exists
+  # and we should just ignore the request.
+  #
+  scope :matching_desired_new_name, lambda { |parsed_name|
+    if parsed_name.rank == "Group"
+      where(search_name: parsed_name.search_name)
+    elsif parsed_name.author.empty?
+      where(text_name: parsed_name.text_name)
+    else
+      where(text_name: parsed_name.text_name).
+        where(author: [parsed_name.author, ""])
+    end
+  }
+  scope :matching_search_name,
+        ->(search_name) { where(search_name: search_name) }
   ### Module Name::Taxonomy
-  scope :with_classification_like,
+  scope :with_rank,
+        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  scope :with_rank_below,
+        ->(rank) { where(Name[:rank] < Name.ranks[rank]) }
+  scope :with_rank_classification_like,
         # Use multi-line lambda literal because fixtures blow up with "lambda":
         # NoMethodError: undefined method `ranks'
         #   test/fixtures/names.yml:28:in `get_binding'
         ->(rank, text_name) { # rubocop:disable Style/Lambda
           where(Name[:classification].matches("%#{rank}: _#{text_name}_%"))
         }
-
-  scope :with_name_like,
+  scope :with_name_like, # Note small diff w :text_name_like scope
         ->(text_name) { where(Name[:text_name].matches("#{text_name} %")) }
-
-  scope :with_rank_below,
-        ->(rank) { where(Name[:rank] < Name.ranks[rank]) }
+  ### Pattern Search
+  # TODO: Maybe refactor tests and Name::Taxonomy to use these methods?
+  # Note small differences between :text_name_like, :with_name_like %
+  scope :classification_like,
+        lambda { |classification|
+          where(Name[:classification].matches("%#{classification}%"))
+        }
+  scope :with_classification, -> { where(Name[:classification].not_blank) }
+  scope :without_classification, -> { where(Name[:classification].blank) }
+  scope :text_name_like,
```

Comment:
Ditto, would prefer "text_name_includes".

Reply:
Question: does it make more sense to make these with/without scopes a single scope and pass a boolean value?

```ruby
scope :with_classification, ->(boolean) {
  if boolean
    where(Name[:classification].not_blank
  else
    where(Name[:classification].blank)
  end
}
```
`Name.with_classification(true)`
Or is that more confusing to write/read, implying that the param is the classification...

Reply:
Leaning against this idea...

Reply:
Agreed.  I think the only use case that really can take advantage of your alternative (combine the two and add a boolean parameter) is if the caller also has a boolean parameter.  But I think the majority of the time, the caller is simply going to want one or the other, and all of those are going to read much better with the two separate scopes.